src/ethereum_test_exceptions/exception_mapper.py

```diff
@@ -76,16 +91,19 @@ def mapper_validator(v: str, info: ValidationInfo) -> Dict[str, Any] | Undefined
     """
     if v is None:
         return v
-    assert isinstance(info.context, dict), f"Invalid context provided: {info.context}"
+    if not isinstance(info.context, dict):
+        return UndefinedException(v, mapper_name="UndefinedExceptionMapper: No context")
     exception_mapper = info.context.get("exception_mapper")
+    if exception_mapper is None:
```

Comment:
exception_mapper is a property of exception? 
where it supposed to be set?

Reply:
It's part only of `ExceptionWithMessage` which is only used by pydantic on types where we parse an exception from t8n or the client itself.

But the types are currently defined where they are used, see e.g.: https://github.com/ethereum/execution-spec-tests/blob/54275170699538a74729197bffd1b05d6ea97208/src/ethereum_test_rpc/types.py#L89-L94

Perhaps we should bring all these types to this file in order to, 1) not redefine, 2) make usage clearer when viewing this file.

Reply:
It's confusing. Message is not part of an exception.

There is ExceptionCode
And error message returned by the tool.
And test validator compares exception code against error message returned by the tool using exception mapper